enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,483 @@
+
+### Workflow Description
+
+There is no supported workflow for this enhancement; when the relevant feature gate is enabled, this functionality works transparently without cluster administrator involvement.
+
+For Red-Hat-internal CI and QE, the agent configuring the install-config needs to set `internalTestingImagePolicy: Disable`.
+The new property can be set regardless of version or whether the `ImagePolicy` feature gate is enabled, because [the installer does not fail on unrecognized properties][install-config-unrecognized-properties].
+
+### API Extensions
+
+As part of implementing this policy, we will drop the following Godocs from [ClusterImagePolicy][ClusterImagePolicy-scope-restriction] and [ImagePolicy-scope-restriction]:
+
+```golang
+// Please be aware that the scopes should not be nested under the repositories of OpenShift Container Platform images.
+// If configured, the policies for OpenShift Container Platform repositories will not be in effect.
+```
+
+[The backing implementation][container-runtime-scope-restriction] will also be removed.
+This guard was [initially intended][scope-restriction-motivation] to protect cluster adminstrators from breaking their clusters by configuring policies that blocked critical images.
+And before Red Hat was publishing signatures for `quay.io/openshift-release-dev/ocp-release` releases, that made sense.
+But now that Red Hat is [almost][OTA-1267] publishing Sigstore signatures for those release images, it makes sense to allow policies covering those images.
+And even if a cluster administrator creates a policy that blocks critical image pulls, PodDisriptionBudgets should keep the Kubernetes API server and related core workloads running for long enough for the cluster administrator to use the Kube API to remove or adjust the problematic policy.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+HyperShift clusters run pods [on the management cluster and in the hosted cluster][hypershift-concepts].
+A ClusterImagePolicy applied to the hosted Kubernetes API server will only be distributed to hosted nodes, and none of those nodes are likely to run an image from `quay.io/openshift-release-dev/ocp-release` (platform DaemonSets will use `quay.io/openshift-release-dev/ocp-v4.0-art-dev` images referenced from the `quay.io/openshift-release-dev/ocp-release` image).
+However, a ClusterImagePolicy applied to the management cluster's Kubernetes API server would cover both the management cluster's cluster-version operator (CVO), and the CVOs associated with hosted clusters (which run as pods in mangement-cluster namespaces).
+That means that it would take an ImagePolicy to be able to distinguish between constraints on the management cluster's CVO and constraints on the hosted components running on the management cluster, which is [out of scope for this enhancement](#use-an-imagepolicy-for-openshift--namespaces).
```

Comment:
HCP should be in scope for this for GA so we need to come up a design that works there.

Reply:
The HyperShift controllers (e.g. HostedCluster controller) could drive an ImagePolicy in the hosted-cluster namespace.  But there's no ImagePolicy yet in 4.17.  And I'm not aware of public guesses at ImagePolicy going GA.  Thoughts on whether that's sufficient for HyperShift, or whether the ImagePolicy timing needs accelerating to hit HyperShift Sigstore timeline targets?

Reply:
Even `ImagePolicy` seems to leave gaps for HCP. The primary attack I worry about is a malicious actor who can indirectly (social engineering/gitops) change HostedCluster/NodePool/ClusterVersion to an untrusted release payload.
For conventional standalone cluster, an `ImagePolicy` in `openshift-cluster-version` with scope "*" would prevent this attack.
For HCP, the `HostedCluster` namespace, which is not `openshift-*`, could contain anything the management cluster administrator wanted (e.g. a custom pod they deploy in each namespace for additional monitoring / management doesn't seem farfetched). 
A "*" scope restricts this. 

We could potentially ask the HyperShift Operator to refuse to act on any HostedCluster/NodePool release payload target that did not start with "quay.io/openshift-release-dev/ocp-release/" .

Reply:
Is there a way to extend the image checks beyond "openshift-*" this way we don't have to treat HCPs are special workload which will need HO education. If something is deemed "bad act" for the management cluster, why should HCPs be an exception (current simplified conjecture, which could be naive).